youtube_dl/extractor/bandlab.py

```diff
@@ -0,0 +1,191 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+
+class BandlabIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bandlab\.com/post/(?P<id>[^/]+)'
+
+    _TEST = {
+        'url': 'https://www.bandlab.com/post/f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+        'info_dict': {
+            'id': 'f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+            'ext': 'm4a',
+            'title': 'ON MY OWN (unreleased)',
+            'artist': 'Michael MacDonald',
+        },
+    }
+
+    def _real_extract(self, url):
+        track_id = self._match_id(url)
+        config = self._download_json(
+            'http://www.bandlab.com/api/v1.3/posts/%s' % track_id, track_id)
+        track_url = config['track']['sample']['audioUrl']
+
+        return {
+            'id': track_id,
+            'title': config['track']['name'],
+            'url': track_url,
+            'artist': config['creator']['name']
```

Comment:
Not mandatory:
```suggestion
            'artist': try_get(config, lambda x: x['creator']['name'].strip(), compat_str),
```
Also could define in this class
```py
    @staticmethod
    def name_or_none(conf):
        return try_get(conf, lambda x: x['creator']['name'].strip(), compat_str)

```
Then above use
```py
            'artist': self.name_or_none(config),
```
and in other classes use `BandcampIE.name_or_none(config)` (or make `BandcampPlaylistIE` inherit from `BandcampIE` and call the method `_name_or_none()`).

Reply:
Whereas `artist` isn't a required field. If the site doesn't send a valid `conf['creator']['name']`, the extractor can return `None` for that field and shouldn't crash. So `try_get()` or similar must be used.